File: Python/_409_Longest_Palindrome/solution.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        hashmap = {}
        for c in s:
            if c not in hashmap:
                hashmap[c] = 1
            else:
                hashmap[c] += 1
        res = 0
        single = 0
        for key in hashmap:
            if hashmap[key] % 2 == 0:
                res += hashmap[key]
            else:
                res += hashmap[key] - 1
                single = 1
        return res + single

File: Java/_409_Longest_Palindrome/Solution.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> map = new HashMap<Character, Integer>();
		for (int i = 0; i < s.length(); i++) {
			if (map.containsKey(s.charAt(i)))
				map.put(s.charAt(i), map.get(s.charAt(i)) + 1);
			else map.put(s.charAt(i), 1);
		}
		int	res = 0;
		int single = 0;
		Iterator it = map.keySet().iterator();  
        while(it.hasNext()) {  
            char key = (char)it.next();
            if (map.get(key) % 2 == 0)
                res += map.get(key);
            else {
                res += map.get(key) - 1;
                single = 1;
            }
        }
		return res + single;
    }
}